### ISSUE TITLE
ENH: Support Hyper-parameter Tuning

### ISSUE THREAD

**Issue by csmcallister** [GSA/srt-ml#3]
As a data scientist, I want to use hyper-parameter tuning so that I can find an optimal model. 
  
## Acceptance criteria
- [x] The tuning job needs to accept at least one estimator.
- [x] The tuning job needs to allow the user to define the parameter space.
- [x] The tuning job needs to allow the user a choice in the method for searching or sampling candidates (e.g. grid search, randomized search, bayesian).
- [x] The tuning job needs to let the user specify a cross-validation scheme, e.g. k-fold.
- [x] The tuning job needs to let the user specify a score function (e.g. accuracy).

**Comment by csmcallister** [comment]
Closed by 298b8e1, which implements a cutomized sklearn pipeline with a randomized grid search.
